config.sample.php

```diff
@@ -125,6 +125,79 @@ class Config
 	 */
 	public static $variants=array(1=>'Classic', 2=>'World', 9=>'AncMed',19=>'Modern2',20=>'Empire4');//3=>'FleetRome', 4=>'CustomStart', 5=>'BuildAnywhere');
 
+	/**
+	 * An array of categories to use when calculating GhostRatings
+	 * @var array
+	 */
+	public static $grCategories=array(
+		/* A Category ID maps to an array of settings */
+		0 => array(
+			/*gives the name of the category*/
+			"name" => "Overall",
+			/*Different scoring systems are used for 1v1 and non-1v1*/
+			"1v1" => "No",
+			/*variantMods sets which variants to include*/
+			"variants" => array(1,2,9,19,20,17),
+			/*pressMods sets which press type to include*/
+			"presses" => array('Regular','PublicPressOnly','NoPress','RulebookPress'),
+			/*phases sets whether you want to include live games, non-live games, or both. The cutoff is at 1 hour phase lengths*/
+			"phases" => array('Live', 'Nonlive'),
+			/*scoring lets you choose what type of scoring sytems to include - only these three types are supported in the current code*/
+			"scoring" => array('Winner-takes-all','Points-per-supply-center','Sum-of-squares')
+		),
+		1 => array(
+			"name" => "Gunboat",
+			"1v1" => "No",
+			"variants" => array(1,2,9,19,20,17),
+			"presses" => array('NoPress'),
+			"phases" => array('Live', 'Nonlive'),
+			"scoring" => array('Winner-takes-all','Points-per-supply-center','Sum-of-squares')
+		),
+		2 => array(
+			"name" => "Live",
+			"1v1" => "No",
+			"variants" => array(1,2,9,19,20,17),
+			"presses" => array('Regular','PublicPressOnly','NoPress','RulebookPress'),
+			"phases" => array('Live'),
+			"scoring" => array('Winner-takes-all','Points-per-supply-center','Sum-of-squares')
+		),
+		3 => array(
+			"name" => "Full Press",
+			"1v1" => "No",
+			"variants" => array(1),
+			"presses" => array('Regular'),
+			"phases" => array('Nonlive'),
+			"scoring" => array('Winner-takes-all','Sum-of-squares')
```

Comment:
Points Per Supply Center is included in each of the categories except for this one. This is to maintain the current categories. In the current scripts, PPSC is not included in this category while it is included in the other categories (I just double checked to make sure this is the case). The settings specified in the config match the settings in the current script.

Reply:
Right but I think the reason for that is the current script only calculates 1 month's worth. So when the only option on the site was PPSC, this category did include it. When it was WTA and PPSC only I also think it included it, it wasn't until PPSC was turned off that this category would have stopped. So to recreate this data via all games I think it has to be on.

Reply:
The current script doesnt just run 1 month, thats why it is so slow. The way that GR is currently run, this specific category purposefully excludes all PPSC games in the history of webdip. If you want to change our categories to include PPSC, then we can do that, but if we are trying to maintain what the current GR scripts do, then this is how it is done. I can show you in the actual scripts if you would like.

Reply:
Sorry, I don't think I'm explaining this well. What I mean is that if you look at when ATC introduced SOS and removed PPSC in the code history it was in 2015 https://github.com/kestasjk/webDiplomacy/commits/5df717f1f7e166a5f417a58c37484d20a5cbe5cc/objects/scoringsystem.php

but if you go to the GR site https://sites.google.com/view/webdipinfo/ghost-ratings/ghost-ratings-by-category this category started in 2012. Back then it was calculated via some other criteria. What I'm trying to figure out is if it was just WTA or PPSC back then. If it was just WTA I think this will translate fine. But if it was PPSC then the historical data we calculate won't even be in the ballpark of what the current old GR monthly files show.

Reply:
So I looked through some users in 2012 and which games count for this category. I confirmed that PPSC games were not counted towards this category in 2012. Only DSS games originally counted towards this category, and was later extended to include SOS when that was later introduced. So this should translate okay.

Reply:
Okay cool, thanks for digging into that